### ISSUE TITLE
[Bug]:  File Versioning Not Working As Expected

### ISSUE THREAD

**Issue by alperozturk96** [nextcloud/server#44978]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

A different file version is not saved for each change made to the file. Therefore the user cannot use different file versions. For example, the user created a file and changed that file every day, at the end of 30 days, when we look at the versions of that file, 30 different file versions do not appear. 

Note: "versions_retention_obligation" -> "disabled"

### Steps to reproduce

1. Create file
2. Edit that file in different time periods such as 1 min later 1 hour later 2 hours later 3 hours later and 3 days later
3. You can't able to see all file versions

### Expected behavior

The file version should be saved according to the desired range.

### Installation method

None

### Nextcloud Server version

27

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by artonge** [comment]
@alperozturk96 are the versions not created, or are they deleted by the expire process?

**Comment by alperozturk96** [comment]
> @alperozturk96 are the versions not created, or are they deleted by the expire process?

I don't have idea about expire process. In my local environment, versions are not created properly as well. I didn't set any expiration limit.

**Comment by artonge** [comment]
> versions are not created properly

So when you edit the file and check the list of versions straight away, there is no new version created?

**Comment by alperozturk96** [comment]
> > versions are not created properly
> 
> So when you edit the file and check the list of versions straight away, there is no new version created?

Nope. I created file. 1 min later I update same file its okay. Then 3 minute later I did same thing a new file version is created as well no problem. 3 hours later again no problem. But at some point if you keep edit same file, a new file version is not exist for that specific time, its just skipping it.

**Comment by alperozturk96** [comment]
> And no specific logs?

[nextcloud.log.zip](https://github.com/nextcloud/server/files/15074381/nextcloud.log.zip)
